src/main/java/org/codevillage/CompositionParsingStep.java

```diff
@@ -0,0 +1,88 @@
+package org.codevillage;
+
+import com.github.javaparser.ast.CompilationUnit;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompositionParsingStep extends EntityParsingStep {
+
+    public CompositionParsingStep(EntityParsingChain next) {
+        super(next);
+    }
+
+    @Override
+    public JavaEntity construct(EntityBuilder builder, CompilationUnit declaration) {
+        // Construct the entity using the builder
+        JavaEntity entity = builder.build();
+
+        // Get the compositions using reflection
+        List<String> compositions = getCompositionClassNames(entity);
+        for (String composition : compositions) {
+            builder.addComposition(composition);
+        }
+
+        // Proceed with the next step in the chain, if any
+        if (next != null) {
+            return next.construct(builder, declaration);
+        }
+        return entity;
+    }
+
+    private List<String> getCompositionClassNames(JavaEntity entity) {
```

Comment:
This... doesn't work. You are getting the *literal class instance* of the JavaEntity and iterating through it. All these steps work around using the CompilationUnit that is passed in to determine things about the JavaEntity that we are trying to build. Please refer to how the other EntityParsingSteps work to see what I mean, the pattern will be similar to LOCParsingStep

Reply:
To clarify, JavaEntity is just a data class. We are calling methods on the builder as we find things out about the entity that we are building from the CompilationUnit (ex. name, realizations, etc). I'd recommend syncing-up with the rest of the repository and looking at the UML to see how everything works together.